src/analyzing/AST_to_IL.ml

```diff
@@ -685,6 +685,52 @@ and expr_aux env ?(void = false) e_gen =
         | _ -> exp
       in
       ident_function_call_hack exp
+  (* x = ClassName(args ...) in Python *)
+  (* ClassName has been resolved to __init__ by the pro engine. *)
+  (* Identified and treated as x = New ClassName(args ...) to support
+     field sensitivity. *)
+  | G.Assign
+      ( ({
+           e =
+             ( G.N (G.Id ((_, _), lhs_info))
+             | G.DotAccess (_, _, FN (Id ((_, _), lhs_info))) );
+           _;
+         } as lhs_e),
+        _,
+        ({
+           e =
+             G.Call
+               ( {
+                   e =
+                     ( G.N (Id (_, id_info))
+                     | G.DotAccess (_, _, FN (Id (_, id_info))) );
+                   _;
+                 },
+                 args );
+           _;
+         } as origin_exp) )
+  (* Can we be tighter about side conditions here to avoid catching
+     direct calls to __init__ ? *)
+    when match id_info.G.id_resolved.contents with
```

Comment:
a match when match ... is a bit heavy. You probably want to move this code in an helper function like
is_python_init_init (),  or even is_constructor_call()

Reply:
Totally agree with Pad here!